src/Integration/src/IntegrationBase/Handler/MethodInvokingMessageProcessor.cs

```diff
@@ -0,0 +1,162 @@
+using Steeltoe.Integration.Support;
+using Steeltoe.Integration.Support.Converter;
+using Steeltoe.Integration.Util;
+using Steeltoe.Messaging;
+using Steeltoe.Messaging.Converter;
+using Steeltoe.Messaging.Handler.Attributes.Support;
+using Steeltoe.Messaging.Handler.Invocation;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Steeltoe.Integration.Handler
+{
+    public class MethodInvokingMessageProcessor<T> : AbstractMessageProcessor<T>, ILifecycle
+    {
+        private readonly object _targetObject;
+        private readonly MethodInfo _method;
+        private readonly IMessageHandlerMethodFactory _messageHandlerMethodFactory;
+        private readonly IInvocableHandlerMethod _invocableHandlerMethod;
+        private IConversionService _conversionService;
+        private bool _isRunning = false;
+
+        public MethodInvokingMessageProcessor(IApplicationContext context, object targetObject, MethodInfo method)
+            : base(context)
+        {
+            _targetObject = targetObject;
+            _method = method;
+            _messageHandlerMethodFactory = ConfigureMessageHandlerFactory();
+            _invocableHandlerMethod = _messageHandlerMethodFactory.CreateInvocableHandlerMethod(_targetObject, _method);
+        }
+
+        public MethodInvokingMessageProcessor(IApplicationContext context, object targetObject, Type attribute)
+            : base(context)
+        {
+            _targetObject = targetObject;
+            _method = FindAnnotatedMethod(targetObject, attribute);
+            _messageHandlerMethodFactory = ConfigureMessageHandlerFactory();
+            _invocableHandlerMethod = _messageHandlerMethodFactory.CreateInvocableHandlerMethod(_targetObject, _method);
+        }
+
+        public virtual IConversionService ConversionService
+        {
+            get
+            {
+                if (_conversionService == null)
+                {
+                    _conversionService = IntegrationServices.ConversionService;
+                }
+
+                return _conversionService;
+            }
+
+            set
+            {
+                _conversionService = value;
+            }
+        }
+
+        public bool IsRunning => _isRunning;
+
+        public Task Start()
+        {
+            _isRunning = true;
+            return Task.CompletedTask;
+        }
+
+        public Task Stop()
+        {
+            _isRunning = false;
+            return Task.CompletedTask;
+        }
+
+        public override T ProcessMessage(IMessage message)
+        {
+            try
+            {
+                var result = _invocableHandlerMethod.Invoke(message);
+#pragma warning disable S2219 // Runtime type checking should be simplified
+                if (result != null && typeof(T).IsAssignableFrom(result.GetType()))
+#pragma warning restore S2219 // Runtime type checking should be simplified
```

Comment:
```suggestion
                if (result is not null and T)
```
Are you sure you want to suppress here?

Reply:
@dtillman can probably answer this for sure, but I am not confident in the S2219 suggestion - which results in the below and is not the same - there might be a bug in that rule?
```csharp   
                if (result is T)
                {
                    return (T)ConversionService.Convert(result, result?.GetType(), typeof(T));
                }
                else
                {
                    return (T)result;
                }
```

Reply:
I do not know if "and T" is equivalent to "&& typeof(T).IsAssignableFrom(result.GetType())" and don't have the time to prove to myself that it is or is not ... :-) ..